plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -146,6 +147,89 @@ fu_plugin_thunderbolt_is_host (GUdevDevice *device)
 	return g_str_has_prefix (name, "domain");
 }
 
+static gchar *
+fu_plugin_thunderbolt_get_force_power_path (FuPlugin *plugin)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+
+	GList *devices;
+	const gchar *basepath;
+	g_autofree gchar *built_path = NULL;
+
+	devices = g_udev_client_query_by_subsystem (data->udev, "wmi");
+	for (GList *l = devices; l != NULL; l = l->next) {
+		GUdevDevice *device = l->data;
+		basepath = g_udev_device_get_sysfs_path (device);
+		built_path = g_build_path ("/", basepath,
+					   "force_power", NULL);
+		if (g_file_test (built_path, G_FILE_TEST_IS_REGULAR))
+			break;
+		built_path = NULL;
```

Comment:
Is this triggers the free from `g_autofree`? I assume not. So an explicit `g_free()` is needed here.
Maybe you want to have a local `g_autofree` instead and if the path test succeeded to find the file, `g_steal_pointer()` it to the outer pointer.

Reply:
I also think this is a confusing function. Maybe we could just use `G_DEFINE_AUTOPTR_CLEANUP_FUNC` to define a function that frees the list and unrefs the contents. Then you could just do `g_steal_pointer (&built_path)` in the exists case and rely on the autofree otherwise.

Reply:
Take a look what I did in the commit I just pushed.  I think this should be clearer and take care of the potential memory leak.